README.md

```diff
@@ -46,18 +49,31 @@ can pipe into it or write to it and call `.end()` when you're done.
 
 ### var epub.write(*obj*, *callback*)
 
-This is the usual stream write function. The object must have the following keys:
+This is the usual stream write function.
 
-* **chapterName** - The name of the chapter in the index.
-* **content** - The HTML content of this chapter.  This will be passed
+#### Adding a chapter
+The object must have the following keys:
+* **`chapterName`** - The name of the chapter in the index.
+* **`content`** - *(optional)* The HTML content of this chapter.  This will be passed
   through `htmltidy` in order to make it valid XHTML.
-* **index** - *(optional)* Where the chapter should show up in the index. These numbers
+* **`sourceFileName`** - *(optional)* Path to the source HTML file with chapter
+* **`index`** - *(optional)* Where the chapter should show up in the index. These numbers
   can have gaps and are used for ordering ONLY. Duplicate index values will
   result in the earlier chapter being excluded from the index. If not specified will
   be added after any where it _was_ specified, in the order written.
-* **fileName** - *(optional)* The filename to use *inside* the epub. This only matters if
+* **`fileName`** - *(optional)* The filename to use *inside* the epub. This only matters if
   you want to do links between chapters. This should end in `.xhtml`.
 
+(`sourceFilename` or `content` should be set)
+
+#### Adding other type of assets
+You can add other types of files to the epub. For example images or stylesheets.
+
+The object must have the following keys:
+* **fileName** - The filename to use *inside* the epub.
+* **sourceFileName** - Path to the source file.
+
```

Comment:
Yeah I agree with you once again. But I don't really like this approach normally. I would rather supply an in between object here. For example a `Chapter`, `Asset` or `CoverImage` object with some required fields.

Not sure on syntax yet but something like this might work:
```
var chapter = new Chapter(title, content, index);
chapter.tidyHtml = true;
streampub.write(chapter);

var asset = new Asset(filename, content);
streampub.write(asset);

var coverImage = new CoverImage(content);
coverImage.autoGenerateHtml = true;
streampub.write(coverImage);
```

This approach would allow us a clean separation of logic (like auto generation of cover HTML, or tidying content).